data-prepper-expression/src/main/antlr/DataPrepperExpression.g4

```diff
@@ -17,16 +17,33 @@ expression
     ;
 
 stringExpression
-    : function
+    : stringExpression PLUS stringExpression
+    | function
     | jsonPointer
     | String
     ;
 
 arithmeticExpression
+    : arithmeticExpression (PLUS | SUBTRACT) arithmeticTerm
+    | arithmeticTerm
+    ;
+
+arithmeticTerm
```

Comment:
Why do you call this a term?

Reply:
I got the following grammar from googling
```
<Exp> ::= <Exp> + <Term> |
<Exp> - <Term> |
<Term>

<Term> ::= <Term> * <Factor> |
<Term> / <Factor> |
<Factor>

<Factor> ::= x | y | ... |
( <Exp> ) |
- <Factor>
```

Reply:
I think we should probably find another name. A term is more generic than two numbers multiplied together.

https://simple.wikipedia.org/wiki/Term_(mathematics)

There might be better names, but perhaps you can split this into a "factor term" and a "summed term?"